src/System.Net.HttpListener/src/System/Net/HttpListenerResponse.cs

```diff
@@ -17,6 +17,11 @@ public WebHeaderCollection Headers
             get => _webHeaders;
             set
             {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
```

Comment:
Why was this necessary?  Doesn't both the desktop and the Windows implementation null ref in the same situation?  At this point let's avoid making unnecessary behavioral changes like this.  We should stick to only adding tests that pass on Windows and then fixing the managed implementation to pass.  If we find bad bugs in the Windows implementation as well, we could look to fix those, but this wouldn't count as a bad bug.